### DISCUSSION TITLE
Find route to NPC from current room

### DISCUSSION THREAD

**Discussion by unix5** [scandum/tintin#28] [Script help]
A quest requires me to visit an NPC whose name has been given. I want to find the way from the current room to the room where the NPC is standing, and save the way as a variable, such as $my_route. I'd like use Map to do this: I write NPC's name as ROOMNOTE of a room. For example, now the quest tolds me the name of the NPC I should visit is Beck. I can do this:

`#var {npc_name} {Beck};`
`#map list {roomnote}{$npc_name}{variable}{npc_room_vnum}; // get the roomvnum of the NPC's room.`
`#map find $npc_room_vnum; // find the shortest path from my current room to the NPC's room and store it in #path.`
`#path save forward my_route; // save the path to variable $my_route.`

That works well. But if there are more than 1 NPC in the same room, it will be different. For example, the ROOMNOTE of a room is:
`{1}{Anie}{2}{Old Fisherman}`
If the quest requires me to visit Old Fisherman, how to get the route and save it in $my_route? Or is there any better way?
Thanks!

**Comment by scandum** [comment]
You could try:
```
#map list {roomnote}{%*\b$npc_name\b%*}{variable}{npc_room_vnum}; // get the roomvnum of the NPC's room.
```

**Comment by unix5** [comment]
`#map list {roomnote}{%*\b$npc_name\b%*}{variable}{npc_room_vnum};`
It doesn't work. But if delete "\b", as:
`#map list {roomnote}{%*$npc_name%*}{variable}{npc_room_vnum};`
It works.
Thanks.
